orders-war/gulpfile.js

```diff
@@ -3,28 +3,26 @@
 var gulp = require('gulp');
 var path = require('path');
 var replace = require('gulp-replace');
-var del = require('del');
+var filter = require('gulp-filter');
 
-gulp.task('web-template-release-Clean', function () {
-    return del(['./target/generated-sources/src/main/resources/templates/web-template-release/**'], {force:true});
-});
 
-gulp.task('web-template-release-Install', function(){
+function webTemplateReleaseInstall () {
+    var htmlFilter = filter('**/*.html', {restore: true});
     return gulp.src([
-        path.resolve('./node_modules/web-template-release/template-cdn-ssi/assets/**/*'),
+        path.resolve('./node_modules/web-template-release/template-cdn-ssi/**/*'),
         path.resolve('./node_modules/web-template-release/package.json')
     ], {base:"./node_modules/web-template-release/template-cdn-ssi/assets/includes-cdn"})
         // See http://mdn.io/string.replace#Specifying_a_string_as_a_parameter
-        .pipe(replace(/GTM-[A-Za-z0-9]{6,10}/g, '${analyticsGtmKey}'))
+        .pipe(replace(/(GTM[A-Za-z0-9-]+)/g, '${analyticsGtmKey}'))
         .pipe(replace(/<!--#include virtual="\/assets\/includes-cdn\/([a-zA-Z-]+)\/(.*)"-->/g, '<#include "..\/$1\/$2"\/>'))
         .pipe(replace(/<!--#include virtual="(.*)"-->/g, '<#include "$1"/>'))
-        .pipe(replace(/(test-static|static)\.qgov\.net\.au/g, "${cdnEnvironment}"))
-        .pipe(replace(/<script( )src/g, "<script nonce=\"${__csp_nonce}\" src"))
-        .pipe(replace(/<link( )href/g, "<link nonce=\"${__csp_nonce}\" href"))
-        .pipe(replace(/<script>/g, "<script nonce=\"${__csp_nonce}\">"))
-        .pipe(gulp.dest('./target/generated-sources/src/main/resources/templates/web-template-release'));
-});
+        .pipe(replace(/(test-static|static)\.qgov\.net\.au/g, '${cdnEnvironment}'))
+        .pipe(htmlFilter).pipe(replace(/<(script|link|style)/g, '<$1 nonce="${__csp_nonce}"'))
```

Comment:
I'm not sure if this should also remove nonce where it is not being used?
From testing on eRTI, I found that the nonce for a \<link\> in a \<noscript\> element and a commented-out \<script\> were present that didn't clear the nonce values.
Could possibly add something like this:
`.pipe(replace(/(<noscript>\n?\s*<link) nonce="\$\{__csp_nonce}"/g, '$1'))`
`.pipe(replace(/(<!--<script) nonce="\$\{__csp_nonce}"/g, '$1'))`

Reply:
Commented out scripts don't run. They should be cleaned up in the long run.

Reply:
good pick up, a better regex is to not introduce it in the first place.
by including ``(?<!<!--|<noscript>\n?\s*)`` prior to the current script.
```
.pipe(htmlFilter).pipe(replace(/(?<!<!--|<noscript>\n?\s*)<(script|link|style)/g, '<$1 nonce="${__csp_nonce}"'))
       
```